libbeat/kibana/client.go

```diff
@@ -132,13 +136,28 @@ func NewClientWithConfig(config *ClientConfig) (*Client, error) {
 		return nil, err
 	}
 
+	// check to see if we have a proxy_host set
+
+	proxyEnv := httpproxy.FromEnvironment()
+	if config.ProxyHost != "" {
+		proxyEnv.HTTPSProxy = config.ProxyHost
+		proxyEnv.HTTPProxy = config.ProxyHost
+	}
+	proxy := func(req *http.Request) (*url.URL, error) {
+		if config.ProxyDisable {
+			return nil, nil
+		}
+		return proxyEnv.ProxyFunc()(req.URL)
```

Comment:
nit: `ProxyFunc` is a 'factory'. We should not call it over and over again. we can change init to read like this (consider to introduce a function configuring the 'Proxy'):

```
var proxy func(_ *http.Request) (*url.URL, error)
if !config.ProxyDisable {
  proxyEnv := ...
  urlProxy := proxyEnv.ProxyFunc()
  proxy = func(req *http.Request) (*url.URL, error) { return urlProxy(req.URL) }
}
```